tests/requests/tests.py

```diff
@@ -557,6 +557,26 @@ def test_get_raw_uri(self):
         request = factory.get('/path/with:colons')
         self.assertEqual(request.get_raw_uri(), 'http://evil.com/path/with:colons')
 
+    @override_settings(ALLOWED_HOSTS=['example.com'])
+    def test_get_params(self):
+        factory = RequestFactory(HTTP_HOST='evil.com')
+        request = factory.get('/?foo=bar')
+        self.assertTrue('foo' in request.GET.keys())
+        self.assertEqual(request.GET['foo'], 'bar')
+
+        request = factory.get(u'/?foo=☂')
```

Comment:
We should urlencode the value since it's otherwise not technically a valid request.